validation/src/main/scala/hmda/validation/engine/ts/quality/TsQualityEngine.scala

```diff
@@ -1,21 +1,34 @@
 package hmda.validation.engine.ts.quality
 
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
 import hmda.model.fi.ts.TransmittalSheet
 import hmda.validation.api.ValidationApi
 import hmda.validation.context.ValidationContext
 import hmda.validation.engine.Quality
 import hmda.validation.engine.ts.TsCommonEngine
 import hmda.validation.rules.ts.quality._
 
+import scala.concurrent.{ ExecutionContext, Future }
+
 trait TsQualityEngine extends TsCommonEngine with ValidationApi {
 
-  def checkQuality(ts: TransmittalSheet, ctx: ValidationContext): TsValidation = {
+  def checkQuality(ts: TransmittalSheet, ctx: ValidationContext)(implicit system: ActorSystem, materializer: ActorMaterializer, ec: ExecutionContext): Future[TsValidation] = {
```

Comment:
I'd suggest you use context bounds as you have done in Q012 to condense this code with implicits.